### ISSUE TITLE
Fix ARG referencing another ARG in FROM statement

### ISSUE THREAD

**Issue by kevinleturc** [fabric8io/docker-maven-plugin#1800]
### Description

Building from a Dockerfile like the one below:
```
ARG IMAGE_NAME=alpine
ARG IMAGE_TAG=latest
ARG FULL_IMAGE=${IMAGE_NAME}:${IMAGE_TAG}
FROM ${FULL_IMAGE}
```
results in the error below:
```
Caused by: java.lang.IllegalArgumentException: Illegal group reference
    at java.util.regex.Matcher.appendExpandedReplacement (Matcher.java:1113)
    at java.util.regex.Matcher.appendReplacement (Matcher.java:1043)
    at java.util.regex.Matcher.replaceFirst (Matcher.java:1446)
    at java.lang.String.replaceFirst (String.java:3025)
    at io.fabric8.maven.docker.util.DockerFileUtil.resolveImageTagFromArgs (DockerFileUtil.java:189)
    at io.fabric8.maven.docker.util.DockerFileUtil.extractBaseImages (DockerFileUtil.java:70)
```

This is because `DockerFileUtil.resolveImageTagFromArgs` doesn't recursively resolve the args, and the curly brackets of the inner arg breaks the Pattern replacement.

### Info

* docker-maven-plugin version: 0.44.0
* Maven version (`mvn -v`): 3.9.8
* Docker version: 27.0.3

**Comment by rohanKanojia** [comment]
@kevinleturc : I have released [0.45.0](https://repo1.maven.org/maven2/io/fabric8/docker-maven-plugin/0.45.0/) . Could you please check if everything is working as expected in your project and provide feedback?

**Comment by kevinleturc** [comment]
> @kevinleturc : I have released [0.45.0](https://repo1.maven.org/maven2/io/fabric8/docker-maven-plugin/0.45.0/) . Could you please check if everything is working as expected in your project and provide feedback?

I have checked the fix in our project and everything is working as expected.
